src/drivers/esp32/src/esp_slip.c

```diff
@@ -106,3 +106,14 @@ static void assembleBuffer(uint8_t *sendBuffer, esp_slip_send_packet *senderPack
 
   sendBuffer[9 + senderPacket->dataSize] = 0xC0;
 }
+
+static void clearUart2Buffer(coms_getDataWithTimeout_t getDataWithTimeout)
+{
+  uint8_t c;
+  bool success = true;
+  while (success)
+  {
+    success = getDataWithTimeout(&c, 1);
+  }
+  return;
```

Comment:
For me this is a do-while loop, but I've learnt that I am quite alone in liking them, so take this with a grain of salt :)

```
do {
  success = getDataWithTimeout(&c, 1);
} while(success);
```

Reply:
Fair enough, I guess this is a classic example of where it can be implemented.